tools/bump-validator/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// isDefinitionFile checks if the basename of the file is definition.yaml or collection.yaml
+func isDefinitionFile(file string) bool {
+	basename := filepath.Base(file)
+	return basename == "definition.yaml" || basename == "collection.yaml"
+}
+
+// reduceRelated keeps only the definition files that are siblings or in a parten directory of the given file
+func reduceRelated(file string, files []string) []string {
+	var relatedFiles []string
+	dir := filepath.Dir(file)
+	for _, f := range files {
+		otherDir := filepath.Dir(f)
+		if isDefinitionFile(f) && (strings.Contains(dir, otherDir) || strings.Contains(otherDir, dir)) {
```

Comment:
Maybe use `strings.HasPrefix` to avoid matching `otherDir` in random location in `dir` (edge case)